src/core/annotation.js

```diff
@@ -661,6 +677,63 @@ class WidgetAnnotation extends Annotation {
     return !!(this.data.fieldFlags & flag);
   }
 
+  /**
+   * Parse widget annotation appearance and extract font information.
+   *
+   * @public
+   * @memberof WidgetAnnotation
+   * @param {number} flag - Hexadecimal representation for an annotation
+   *                        field characteristic
+   * @return {Promise}
+   */
+  parseAppearance(params) {
+    if (!params.pdfManager || !params.pdfManager.pdfDocument) {
+      return Promise.resolve(this);
+    }
+
+    let data = this.data;
+    let self = this;
+
+    let opList = new OperatorList();
+    let appearanceStream = new Stream(stringToBytes(data.defaultAppearance));
+    let formFonts = params.pdfManager.pdfDocument.acroForm.get('DR');
+    opList.acroForm = params.pdfManager.pdfDocument.acroForm;
+    return params.evaluator.getOperatorList({
+      stream: appearanceStream,
+      task: params.task,
+      resources: formFonts,
+      operatorList: opList,
+    }).then(() => {
+      let a = opList.argsArray;
+      let i;
```

Comment:
Remove this line and put the `let` in the `for` loop instead: `for (let i = 0, ii = opList.fnArray.length; i < ii; i++) {` is usually how we write those loops (since it caches the length too).

Reply:
Ok, cool, thank you